packages/components/src/__future__/Tag/Tag.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useMemo, FC, ReactNode, Children, isValidElement } from "react"
+import classNames from "classnames"
+import { v4 } from "uuid"
+
+const isElementText = (child: ReactNode): boolean =>
+  typeof child === "string" && child.replace(/\s/g, "").length > 0
+
+const isElementNumber = (child: ReactNode): boolean => typeof child === "number"
+
+const isElementIcon = (child: ReactNode): boolean =>
+  isValidElement(child) &&
+  typeof child.type !== "undefined" &&
+  typeof child.type !== "string" &&
+  typeof child.type !== "number" &&
+  child.type.name === "Icon"
+
+type TagColors =
+  | "gray"
+  | "blue"
+  | "green"
+  | "yellow"
+  | "orange"
+  | "red"
+  | "purple"
+
+export interface TagProps {
+  children: ReactNode
+  color?: TagColors
+}
```

Comment:
Don't worry about adding the icon prop, we will need to do extra work to make it work anyways.